src/codegen/spirv.zig

```diff
@@ -3222,11 +3273,17 @@ const NavGen = struct {
         is_volatile: bool = false,
     };
 
-    fn load(self: *NavGen, value_ty: Type, ptr_id: IdRef, options: MemoryOptions) !IdRef {
+    fn load(self: *NavGen, value_ty: Type, ptr_id: IdRef, storage_class: StorageClass, options: MemoryOptions) !IdRef {
+        const zcu = self.pt.zcu;
+        const alignment: u32 = @intCast(value_ty.abiAlignment(zcu).toByteUnits().?);
         const indirect_value_ty_id = try self.resolveType(value_ty, .indirect);
         const result_id = self.spv.allocId();
         const access = spec.MemoryAccess.Extended{
             .Volatile = options.is_volatile,
+            .Aligned = switch (storage_class) {
+                .PhysicalStorageBuffer => .{ .literal_integer = alignment },
+                else => null,
+            },
```

Comment:
Is it valid to put Aligned on logical pointer loads? If so, I suggest emitting the align unconditionally.

Reply:
i dont think it should be an issue